### PR TITLE
relayer arguments injection

### PR DESCRIPTION
**Pull request by dragos-rebegea**
Arguments injection in relayer component preparing for integration testing

### CODE REVIEW

File: bridge/eth/client.go [hunk 1]
```diff
@@ -61,30 +61,24 @@ type Client struct {
...
-	mapper bridge.Mapper,
-	gasHandler bridge.GasHandler,
-) (*Client, error) {
+// ArgsClient is the DTO used in the client constructor
+type ArgsClient struct {
+	Config      bridge.EthereumConfig
+	Broadcaster bridge.Broadcaster
+	Mapper      bridge.Mapper
+	GasHandler  bridge.GasHandler
+	EthClient   *ethclient.Client
```
Comment (iulianpascalau) [inline bridge/eth/client.go:70]: You used pointers. There is no way we can ever use mocked implementations in integration tests.
Let's switch to:
```GO
EthClient     BlockchainClient
EthInstance BridgeContract
```

valid also for the ArgsRelayer

File: relay/relay.go [hunk 2]
```diff
@@ -99,14 +100,24 @@ type Relay struct {
 	pollingHandlers      []io.Closer
 }
 
-func NewRelay(config Config, flagsConfig ContextFlagsConfig, name string) (*Relay, error) {
+// ArgsRelayer is the DTO used in the relayer constructor
+type ArgsRelayer struct {
+	Config      Config
+	FlagsConfig ContextFlagsConfig
+	Name        string
+	Proxy       bridge.ElrondProxy
```
Comment (iulianpascalau) [inline relay/relay.go:108]: :+1: for the Proxy
The `EthClient` and `EthInstance` should change to interfaces

File: relay/relay.go [hunk 3]
```diff
@@ -136,37 +146,46 @@ func NewRelay(config Config, flagsConfig ContextFlagsConfig, name string) (*Rela
 	}
 	relay.elrondBridge = elrondBridge
 
+	gasStationConfig := args.Config.Eth.GasStation
```
Comment (iulianpascalau) [inline relay/relay.go:149]: :+1:

**Review by iulianpascalau — CHANGES_REQUESTED** [review]

**Review by iulianpascalau — DISMISSED** [review]

**Review by mihaieremia — DISMISSED** [review]

File: bridge/eth/client.go [hunk 4]
```diff
@@ -61,30 +60,24 @@ type Client struct {
...
+	EthClient   BlockchainClient
+	EthInstance BridgeContract
+}
+
+func NewClient(args ArgsClient) (*Client, error) {
 
 	log := logger.GetOrCreate("EthClient")
 
-	if check.IfNil(gasHandler) {
+	if check.IfNil(args.GasHandler) {
```
Comment (raduchis) [inline bridge/eth/client.go:77]: there are no nil checks on other args fields: Config, Broadcaster, Mapper etc. 
same thing also in other files that use args

Comment (dragos-rebegea) [reply bridge/eth/client.go:78]: Yes, indeed we need that. But we can proceed with this one like it is and 1 follow-up PR will come with changes for all missed check in all args types.

Comment (dragos-rebegea) [reply bridge/eth/client.go:78]: [Check args for all args types #105](https://github.com/ElrondNetwork/elrond-eth-bridge/pull/105)

**Review by sasurobert — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Pull request merged**